connectors/memory/memory.go

```diff
@@ -534,15 +534,9 @@ func (c *Connector) Scan(_ context.Context, ei *dosa.EntityInfo, minimumFields [
 
 	// if there was a token, decode it so we can determine the starting
 	// partition key
-	var startPartKey string
-	var start map[string]dosa.FieldValue
-	if token != "" {
-		var err error
-		start, err = decodeToken(token)
-		if err != nil {
-			return nil, "", errors.Wrapf(err, "Invalid token %q", token)
-		}
-		startPartKey = partitionKeyBuilder(ei, start)
+	startPartKey, start, err := getStartingPoint(ei, token)
+	if err != nil {
+		return nil, "", errors.Wrapf(err, "Invalid token %s", token)
```

Comment:
Nit, I think this wrap error message is incorrect. The `getStartingPoint` function could've failed for any number of reasons. I would make this wrap error message something like `failed to scan` and then (if you want) wrap the error from `decodeToken` down in `getStartingPoint`. In _that_ wrap, I would mention about the invalid token.